test/jdk/java/nio/jni/NewDirectByteBuffer.java

```diff
@@ -42,31 +50,49 @@ public class NewDirectByteBuffer {
         5_000_000_000L
     };
 
-    private static final long[] LEGAL_CAPACITIES = {
-        0L,
-        1L,
-        (long)Integer.MAX_VALUE - 1,
-        (long)Integer.MAX_VALUE
-    };
+    private static final void checkBuffer(ByteBuffer buf, long capacity) {
+        if (!buf.isDirect())
+            throw new RuntimeException("Buffer is not direct");
+        long bufferCapacity = getDirectBufferCapacity(buf);
+        if (bufferCapacity != capacity)
+            throw new RuntimeException("GetDirectBufferCapacity "
+                + bufferCapacity + " is not " + capacity);
+        if (buf.capacity() != capacity)
+            throw new RuntimeException("buf.capacity() "
+                + buf.capacity() + " is not " + capacity);
+        if (buf.position() != 0)
+            throw new RuntimeException("buf.position() "
+                + buf.position() + " is nonzero");
+        if (buf.limit() != capacity)
+            throw new RuntimeException("buf.limit() "
+                + buf.limit() + " is not " + capacity);
+    }
 
     public static void main(String[] args) {
         System.out.println("--- Legal Capacities ---");
         for (long cap : LEGAL_CAPACITIES) {
```

Comment:
I would still expect this to throw OOME for the very large values, so the OOME needs to be caught so that the test continues.

Reply:
Fixed in 82d9e6f1f6782b180943d1136881d58dd2852e58.